Missing #include <cstdint> in some files

The missing includes are causing multiple build errors on Fedora 42 with clang. Trimmed output:

```
/home/rgarcia/devel/amber/src/type.h:55:39: error: ‘uint32_t’ has not been declared
   55 |   static bool IsInt8(FormatMode mode, uint32_t num_bits) {
      |                                       ^~~~~~~~
```

```
/home/rgarcia/devel/amber/src/tokenizer.h:26:6: error: elaborated-type-specifier for a scoped enum must not use the ‘class’ keyword
   26 | enum class TokenType : uint8_t {
      | ~~~~ ^~~~~
      |      -----
```

--- a/src/tokenizer.h
+++ b/src/tokenizer.h
@@ -16,6 +16,7 @@
 #define SRC_TOKENIZER_H_
 
 #include <cstdlib>
+#include <cstdint>
 #include <memory>
 #include <string>
 

--- a/src/type.h
+++ b/src/type.h
@@ -16,6 +16,7 @@
 #define SRC_TYPE_H_
 
 #include <cassert>
+#include <cstdint>
 #include <memory>
 #include <string>
 #include <vector>

